src/account.ts

```diff
@@ -180,10 +180,56 @@ export abstract class Account {
         );
     }
 
+    private async estimateDeclareV2Fee(
+        contractFactory: StarknetContractFactory,
+        options: EstimateFeeOptions = {}
+    ): Promise<starknet.FeeEstimation> {
+        const maxFee = (options.maxFee || 0).toString();
+        const version = DECLARE_VERSION;
+        const nonce = options.nonce == null ? await this.getNonce() : options.nonce;
+
+        const hre = await import("hardhat");
+        const chainId = hre.starknet.networkConfig.starknetChainId;
+
+        let compiledClassHash = await hre.starknetWrapper.getCompiledClassHash(
+            contractFactory.casmPath
+        );
+        compiledClassHash = numericToHexString(BigInt(compiledClassHash));
+        const classHash = await hre.starknetWrapper.getSierraContractClassHash(
+            contractFactory.metadataPath
+        );
+
+        const calldata = [classHash];
+        const messageHash = calculateDeclareV2TxHash(
+            this.address,
+            calldata,
+            maxFee.toString(),
+            chainId,
+            [nonce.toString(), compiledClassHash]
+        );
+
+        const signatures = this.getSignatures(messageHash);
+        const data = {
+            type: "DECLARE",
+            sender_address: this.address,
+            compiled_class_hash: compiledClassHash,
+            contract_class: readCairo1Contract(contractFactory.metadataPath).getCompiledClass(),
+            signature: bnToDecimalStringArray(signatures || []),
```

Comment:
Why this defaulting to `[]`?

Reply:
Agreed to keep it as disscussed outside github.